src/transformers/models/encoder_decoder/modeling_encoder_decoder.py

```diff
@@ -620,6 +620,8 @@ def forward(
             decoder_input_ids = shift_tokens_right(
                 labels, self.config.pad_token_id, self.config.decoder_start_token_id
             )
+            if decoder_attention_mask is None:
+                decoder_attention_mask = decoder_input_ids.new_tensor(labels != self.config.pad_token_id)
```

Comment:
```suggestion
                decoder_attention_mask = decoder_input_ids.new_tensor(decoder_input_ids != self.config.pad_token_id)
```
we should create it using the decoder input ids no?

Reply:
Yup that's right, updated to decoder input ids.